ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java

```diff
@@ -0,0 +1,154 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.model.InstanceConfig;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * A class with clustermap related utility methods for use by other classes.
+ */
+public class ClusterMapUtils {
+  static final String DISK_CAPACITY_STR = "capacityInBytes";
+  static final String DISK_STATE = "diskState";
+  static final String REPLICAS_STR = "Replicas";
+  static final String REPLICAS_DELIM_STR = ",";
+  static final String REPLICAS_STR_SEPARATOR = ":";
+  static final String SSLPORT_STR = "sslPort";
+  static final String RACKID_STR = "rackId";
+  static final String SEALED_STR = "SEALED";
+  static final String AVAILABLE_STR = "AVAILABLE";
+  static final String UNAVAILABLE_STR = "UNAVAILABLE";
+  static final String ZKCONNECTSTR_STR = "zkConnectStr";
+  static final String ZKINFO_STR = "zkInfo";
+  static final String DATACENTER_STR = "datacenter";
+  static final int UNKNOWN_RACK_ID = -1;
+  static final int MIN_PORT = 1025;
+  static final int MAX_PORT = 65535;
+  private static final long MIN_REPLICA_CAPACITY_IN_BYTES = 1024 * 1024 * 1024L;
+  private static final long MAX_REPLICA_CAPACITY_IN_BYTES = 10L * 1024 * 1024 * 1024 * 1024;
+  private static final long MIN_DISK_CAPACITY_IN_BYTES = 10L * 1024 * 1024 * 1024;
+  private static final long MAX_DISK_CAPACITY_IN_BYTES = 10L * 1024 * 1024 * 1024 * 1024;
+
+  /**
+   * Construct and return the instance name given the host and port.
+   * @param host the hostname of the instance.
+   * @param port the port of the instance. Can be null.
+   * @return the constructed instance name.
+   */
+  public static String getInstanceName(String host, Integer port) {
+    return port == null ? host : host + "_" + port;
+  }
+
+  /**
+   * Parses zk layout JSON string and populates and returns a map of datacenter name to Zk connect strings.
+   * @param zkLayoutJsonString the string containing the Zookeeper layout.
+   * @throws JSONException if there is an error parsing the JSON.
+   */
+  static Map<String, String> parseZkJsonAndPopulateZkInfo(String zkLayoutJsonString) throws JSONException {
+    Map<String, String> dataCenterToZkAddress = new HashMap<>();
+    JSONObject root = new JSONObject(zkLayoutJsonString);
+    JSONArray all = (root).getJSONArray(ZKINFO_STR);
```

Comment:
Not needed. I must have at one point had it as
`(new JSONObject(zkLayoutJsonString)).getJSONArray()`.